service/lib/agama/manager.rb

```diff
@@ -205,18 +205,22 @@ def valid?
 
     # Collects the logs and stores them into an archive
     #
-    # @param user [String] local username who will own archive
+    # @param path [String] directory where to store logs
+    #
     # @return [String] path to created archive
-    def collect_logs(user)
-      output = Yast::Execute.locally!("save_y2logs", stderr: :capture)
-      path = output[/^.* (\/tmp\/y2log-\S*)/, 1]
-      Yast::Execute.locally!("chown", "#{user}:", path)
+    def collect_logs(path: nil)
+      opt = "-d #{path}" if !path.nil? && !path.empty?
```

Comment:
np: would not be more readable `unless path.nil? || path.empty?`? Or even `unless path.to_s.empty?`.